src/transports/IframeExecutionEnvironmentTransport.ts

```diff
@@ -0,0 +1,84 @@
+import { Duplex } from 'stream';
+import { Transport } from '@open-rpc/client-js/build/transports/Transport';
+import { IJSONRPCData } from '@open-rpc/client-js/build/Request';
+import { WindowPostMessageStream } from '@metamask/post-message-stream';
+import ObjectMultiplex from '@metamask/object-multiplex';
+import pump from 'pump';
+
+class IframeExecutionEnvironmentTransport extends Transport {
+  private uri: string;
+
+  private stream?: WindowPostMessageStream;
+
+  private commandStream?: Duplex;
+
+  private frame?: Window;
+
+  private postMessageID: string;
+
+  constructor(uri: string) {
+    super();
+    this.uri = uri;
+    this.postMessageID = `post-message-transport-${Math.random()}`;
+  }
+
+  public createWindow(uri: string): Promise<Window> {
+    const iframe = document.createElement('iframe');
+    return new Promise((resolve) => {
+      iframe.addEventListener('load', () => {
+        if (iframe.contentWindow) {
+          resolve(iframe.contentWindow);
+        }
+      });
+      document.body.appendChild(iframe);
+      iframe.setAttribute('src', uri);
+      iframe.setAttribute('id', this.postMessageID);
+    });
+  }
+
+  public async connect(): Promise<boolean> {
+    this.frame = await this.createWindow(this.uri);
+    this.stream = new WindowPostMessageStream({
+      name: 'parent',
+      target: 'child',
+      targetWindow: this.frame,
+    });
+
+    const mux = this.setupMultiplex(this.stream, 'TransportStream');
+
+    this.commandStream = mux.createStream('command') as unknown as Duplex;
+
+    this.commandStream.on('data', (data: unknown) => {
+      this.transportRequestManager.resolveResponse(JSON.stringify(data));
+    });
+    return true;
+  }
+
+  public async sendData(
+    data: IJSONRPCData,
+    timeout: number | undefined = 5000,
+  ): Promise<any> {
+    const prom = this.transportRequestManager.addRequest(data, timeout);
+    this.commandStream?.write((data as IJSONRPCData).request);
```

Comment:
This cast seems unnecessary - it's already declared as `IJSONRPCData`

```suggestion
    this.commandStream?.write(data).request);
```